crates/hir-def/src/nameres/collector.rs

```diff
@@ -1425,7 +1425,20 @@ impl DefCollector<'_> {
             tracing::warn!("macro expansion is too deep");
             return;
         }
-        let file_id = macro_call_id.as_file();
+
+        let loc = macro_call_id.lookup(self.db.upcast());
+        let file_id = if loc.def.is_include() {
+            // If the given macro is an builtin `include!` macro, we expand into the included file.
+            // So we should collect items within it.
+            if let Some(it) = loc.include_file_id(self.db.upcast(), macro_call_id) {
+                it.into()
+            } else {
+                // This case the file is not exist. Do not collect items
+                return;
+            }
+        } else {
+            macro_call_id.as_file()
+        };
```

Comment:
This id mismatch was blocking us from inserting item for usage search

https://github.com/rust-lang/rust-analyzer/blob/8dd53a3a46adffdc7928bbfabab90d6348c9a089/crates/hir-def/src/child_by_source.rs#L260

`LHS: MacroFileId != RHS: EditionFileId`

Reply:
I wonder, does it mean we have to make the `include!()`d file a macro file instead of an `EditionedFileId`?

Reply:
I thought about both directions; making included file into `MacroFile` as you said and current way and I think that as the ide thing triggers from the opened included file are anchored with nodes from `EditionedFile`, this way would be more intuitive. But I'm not 100% sure about this so I'll rethink about this

Reply:
Well, I'm afraid that making it a macro file would require us to map between `InFile<T> { file_id: Editioned, .. }` and `InFile<T> { file_id: Macro, .. }` back and forth quite often and missing it in one place might cause another error. This PR seems simpler